lib/src/dartdoc_options.dart

```diff
@@ -990,10 +990,14 @@ abstract class _DartdocFileOption<T> implements DartdocOption<T> {
             .pathContext
             .join(dir.path, 'dartdoc_options.yaml'));
         if (dartdocOptionsFile.exists) {
-          yamlData = _YamlFileData(
-              loadYaml(dartdocOptionsFile.readAsStringSync()),
-              resourceProvider.pathContext.canonicalize(dir.path));
-          break;
+          final yaml = loadYaml(dartdocOptionsFile.readAsStringSync());
+          // [loadYaml()] will return null for empty (or all comment) yaml files,
+          // so we must check for that case.
+          if (yaml != null) {
+            yamlData = _YamlFileData(
+                yaml, resourceProvider.pathContext.canonicalize(dir.path));
+            break;
```

Comment:
I feel like we should still break, even if `yaml == null`, right? I think once we hit a dartdoc options file, even if its empty, we should stop walking ancestors.

Reply:
Ok, I have changed that so we now break even when yaml==null.